service.autostop/service.py

```diff
@@ -54,66 +59,48 @@ def onPlayBackStopped(self):
             elif player.paflag == 0:
                 pacount = 0
         except:
-            xbmc.log('Autostop pause count and stop time exception error' + str(pacount) + ' ' + str(pastoptime) +    \
-            ' ' + str(player.paflag), xbmc.LOGINFO)            
+            xbmc.log('Autostop pause count and stop time exception error' + str(pacount) + ' ' + \
+            str(pastoptime) + ' ' + str(player.paflag), xbmc.LOGINFO)
+            pass            
 
-    plcount += 1 
-    if plcount % 10 == 0:                          # Check for playing video every 10 seconds
+    plcount = playCount(plcount, padjust, player.plflag, player.paflag, asevlog)
+    sleepNotify(plcount, plstoptime, player.plflag, plnotify, plextend, asevlog, player.paflag) 
+    if plcount % 10 == 0:                            # Check for playing video every 10 seconds
         try:
             plstoptime = int(settings('plstop'))
-            xbmc.log('Autostop play count and stop time ' + str(plcount) + ' ' + str(plstoptime) +    \
-            ' ' + str(player.plflag), xbmc.LOGDEBUG)
-            if plstoptime > 0 and plcount >= plstoptime * 60 and player.plflag == 1:
-                if xbmc.Player().isPlayingVideo():
-                    ptag = xbmc.Player().getVideoInfoTag()
-                    ptitle = ptag.getTitle()
-                elif xbmc.Player().isPlayingAudio():
-                    ptag = xbmc.Player().getMusicInfoTag()
-                    ptitle = ptag.getTitle()                
-                else:
-                    ptitle = "playing file"
-                pos = xbmc.Player().getTime()
-                xbmc.Player().stop()
-                plcount = 0
-                mgenlog ='Autostop stopped current playback: ' + ptitle +     \
-                ' at: ' + time.strftime("%H:%M:%S", time.gmtime(pos))
-                xbmc.log(mgenlog, xbmc.LOGINFO)
-                dialog = xbmcgui.Dialog()
-                dialog.notification('Autostop Sleep Timer', mgenlog, xbmcgui.NOTIFICATION_INFO, 5000)
-                if settings('screensaver') == 'true':  #  Active screensaver if option selected
-                    xbmc.executebuiltin('ActivateScreensaver')
-            elif player.plflag == 0:
+            padjust = settings('padjust')
+            plnotify = int(settings('plnotify'))
+            plextend = int(settings('plextend'))
+            extime = float(settings('extime'))
+            asevlog = settings('asevlog')
+            totalstoptime = plstoptime + extime
+            if asevlog == 'true':
+                xbmc.log('Autostop play count and stop time ' + str(plcount) + ' ' +         \
+                str(plstoptime) + ' ' + str(player.plflag), xbmc.LOGINFO)
+            if plstoptime > 0 and plcount >= totalstoptime * 60 and (player.plflag > 0 or    \
+            (player.plflag == 0 and player.paflag == 1)):
+                logmsg = 'Autostop stopped current playback: '
+                notifymsg = 'Autostop Sleep Timer'
```

Comment:
Same

Reply:
This has been updated to localized strings.